cmd/cloudFoundryDeploy.go

```diff
@@ -0,0 +1,855 @@
+				return "", []string{}, []string{}, err
+			}
+
+			manifestVariablesFiles, err := removeDefaultManifestVariableFileIfItIsTheOnlyFileAndThatFileDoesNotExist(config.ManifestVariablesFiles)
+			if err != nil {
+				return "", []string{}, []string{}, err
+			}
+
+			modified, err := _substitute(config.Manifest, manifestVariables, manifestVariablesFiles)
+			if err != nil {
+				return "", []string{}, []string{}, errors.Wrap(err, "Cannot prepare manifest file")
+			}
+
+			if modified {
+				log.Entry().Infof("Manifest file '%s' has been updated (variable substitution)", config.Manifest)
+			} else {
+				log.Entry().Infof("Manifest file '%s' has not been updated (variable substitution)", config.Manifest)
+			}
+
+			err = handleLegacyCfManifest(config.Manifest)
+			if err != nil {
+				return "", []string{}, []string{}, err
+			}
+		}
+	} else {
+		log.Entry().Info("No manifest file configured")
+	}
+	return "blue-green-deploy", deployOptions, smokeTest, nil
+}
+
+func removeDefaultManifestVariableFileIfItIsTheOnlyFileAndThatFileDoesNotExist(manifestVariablesFiles []string) ([]string, error) {
+
+	const defaultManifestVariableFileName = "manifest-variables.yml"
+	if len(manifestVariablesFiles) == 1 && manifestVariablesFiles[0] == defaultManifestVariableFileName {
+		// we have only the default file. Most likely this is not configured, but we simply have the default.
+		// In case this file does not exist we ignore that file.
+		exists, err := fileUtils.FileExists(defaultManifestVariableFileName)
+		if err != nil {
+			return []string{}, err
+		}
+		if !exists {
+			return []string{}, nil
+		}
+	}
+	return manifestVariablesFiles, nil
+}
+
+func toParameterMap(parameters []string) (*orderedmap.OrderedMap, error) {
+
+	parameterMap := orderedmap.NewOrderedMap()
+
+	for _, p := range parameters {
+		keyVal := strings.Split(p, "=")
+		if len(keyVal) != 2 {
+			return nil, fmt.Errorf("Invalid parameter provided (expected format <key>=<val>: '%s'", p)
+		}
+		parameterMap.Set(keyVal[0], keyVal[1])
+	}
+	return parameterMap, nil
+}
+
+func handleLegacyCfManifest(manifestFile string) error {
+	manifest, err := _getManifest(manifestFile)
+	if err != nil {
+		return err
+	}
+
+	err = manifest.Transform()
+	if err != nil {
+		return err
+	}
+	if manifest.IsModified() {
+
+		err = manifest.WriteManifest()
+
+		if err != nil {
+			return err
+		}
+		log.Entry().Infof("Manifest file '%s' was in legacy format has been transformed and updated.", manifestFile)
+	} else {
+		log.Entry().Infof("Manifest file '%s' was not in legacy format. No tranformation needed, no update performed.", manifestFile)
+	}
+	return nil
+}
+
+func prepareCfPushCfNativeDeploy(config *cloudFoundryDeployOptions) (string, []string, []string, error) {
+
+	deployOptions := []string{}
+	varOptions, err := getVarOptions(config.ManifestVariables)
+	if err != nil {
+		return "", []string{}, []string{}, err
+	}
+	varFileOptions, err := getVarFileOptions(config.ManifestVariablesFiles)
```

Comment:
Is it OK to use here directly, while it is validated in the other function?

Reply:
That came in my mind, too, but I left is basically similar to what I found in groovy. This is basically what I had in mind when proposing to merge soon and to refactor later. Is is easier to comprehend the transition from groovy to go when we try to keep it similar.

Of course I can change that now, if you want ...

By the way: The validation is done somewhere else in the code-flow (was like this it groovy). I guess we have here - like often in that case - some "historical reasons".

Reply:
Now we validate in both cases